src/HttpHeader.cc

```diff
@@ -517,7 +515,13 @@ HttpHeader::parse(const char *header_start, size_t hdrLen)
                Raw("header", header_start, hdrLen));
     }
 
-    if (chunked()) {
+    if (clen.prohibitedAndIgnored) {
+        // RFC 7230 section 3.3.2: A server MUST NOT send a Content-Length
+        // header field in any response with a status code of 1xx (Informational)
+        // or 204 (No Content).
```

Comment:
Please also add something like `And RFC 7230 3.3.3#1 tells recipients to ignore such Content-Lengths`. The existing comment explain why those header fields are prohibited, while the additional comment explains why we are ignoring them.